src/components/PostComment/PostComment.jsx

```diff
@@ -0,0 +1,92 @@
+import "./PostComment.css"
+import { useState, useRef } from "react";
+import { postNewComment } from "../../api";
+import { useParams } from "react-router-dom";
+
+function PostComment(props){
+    const [currComment, setCurrComment] = useState('');
+    const {article_id} = useParams();
+    const textareaRef = useRef(null);
+
+    const [isPostEmpty, setIsPostEmpty] = useState(false);
+    const [isPostSent, setIsPostSent] = useState(false);
+    const [isPostSuccessful, setIsPostSuccessful] = useState(false);
+    const [isPostFailed, setIsPostFailed] = useState(false);
+
+   
+    
+    //stop if button not pressed
+    if(!props.isPostShowing){
+        return null;
+    }
+
+    function formValidator(e){
+        e.preventDefault();
+        
+        setIsPostFailed(false)
+        setIsPostSuccessful(false)
+        setIsPostEmpty(false)
+        //edge case, states still need to be reset
+        if(isPostSent || isPostSuccessful || isPostFailed){
+            return;
+        }
+        //edge case, no text
+        if(currComment === ""){
+            setIsPostEmpty(true);
+            return
+        }
+
+        const commentForSubmitting = e.target[0].value;
+        const commentObj = {
+            username: 'jessjelly',
+            body: commentForSubmitting
+        }
+        
+        setIsPostSent(true);
+
+        postNewComment(article_id, commentObj)
+        .then((posted_article) => {
+            setCurrComment('');//stops repeated comment posting
```

Comment:
we should also make sure we add the newly posted comment to the comments array so that it shows up in the list of comments - we can do this optimistically or we can do this once the request is successful